### PR TITLE
#166214077 Configure Travis CI build

### PR DESCRIPTION
**Pull request by Adekoreday**
#### What does this PR do?

Configure Travis build through .travis.yml

#### Description of Task to be completed?

N/A

#### How should this be manually tested?

By settings, Travis build run when a pull request is made to the develop branch

#### What are the relevant pivotal tracker stories?*

#166214077

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  directories:
+    - "node_modules"
+
+before_script:
+  - psql -c 'create database travis_ci_test;' -U postgres
+
+script: npm run test
+
+after_success:
+  - npm run coverage
```
Comment (allebd) [inline .travis.yml]: New line required at the end of the file
Thanks it has been added

Comment (allebd) [reply .travis.yml]: Not treated accordingly yet, there is an extra space currently included with the new line
treated accordingly thanks

**Review by Lundii — COMMENTED** [review]
Is there a way we can test this PR on our local? Like anything that can give indication that the PR is working as expected

You can test it by viewing the travis build that passed on travis and inspecting the .travis.yml

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Hello-Books-BE
 
-[![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com)
+[![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com) https://travis-ci.org/KvNGCzA/Hello-Books-BE.svg?branch=develop
```
Comment (KvNGCzA) [inline README.md]: The Travis badge should be a `markdown` not a link.
ok thanks
Fixed

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+language: node_js
+node_js:
+  - "stable"
+install: npm install
+
+services:
+  - postgresql
+
```
Comment (KvNGCzA) [inline .travis.yml:8]: Add the codeclimate repo token

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 3 above)
Comment (Adekoreday) [reply .travis.yml:8]: ok seen thanks
don't have access to the codeclimate token

**Comment by Lundii** [comment]
> Is there a way we can test this PR on our local? Like anything that can give indication that the PR is working as expected
> 
> You can test it by viewing the travis build that passed on travis and inspecting the .travis.yml

Okay. Great

**Review by Lundii — DISMISSED** [review]
Is there a way we can test this PR on our local? Like anything that can give indication that the PR is working as expected

File: .travis.yml (hunk 3 above)
Comment (KvNGCzA) [reply .travis.yml:8]: @Adekoreday Here you go `205b460fb18d22d3db847cb98cc3fbbee316a0624d59024b5e15728eb78ef38e`

File: README.md [hunk 4]
```diff
@@ -1,6 +1,8 @@
 # Hello-Books-BE
 
 [![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com)
+[![Build Status](https://travis-ci.org/KvNGCzA/Hello-Books-BE.svg?branch=develop)](https://travis-ci.org/KvNGCzA/Hello-Books-BE)
+
```
Comment (KvNGCzA) [inline README.md:7]: Add this to the readme file `[Maintainability](https://codeclimate.com/github/KvNGCzA/Hello-Books-BE/maintainability)`

**Comment by KvNGCzA** [comment]
remove this `[Delivers #166214077]` from your PR description

**Comment by KvNGCzA** [comment]
Your Travis script is running the tests twice, fix this, please.

Screen Shot 2019-05-29 at 7 11 52 AM

will attend to this
fixed

**Review by allebd — DISMISSED** [review]
Good work

thanks

File: lcov.info [hunk 5]
```diff
@@ -0,0 +1,30 @@
+TN:
```
Comment (KvNGCzA) [inline lcov.info]: Kindly get rid of this file and add it to .gitignore

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

File: lcov.info (hunk 5 above)
Comment (Adekoreday) [reply lcov.info]: done thanks

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+cache:
+  directories:
+    - "node_modules"
+
+env: 
+  - CODECLIMATE_REPO_TOKEN=205b460fb18d22d3db847cb98cc3fbbee316a0624d59024b5e15728eb78ef38e
```
Comment (KvNGCzA) [inline .travis.yml]: is this how this should be done?

**Review by JuwonAbiola — APPROVED** [review]
good work

**Review by Lundii — APPROVED** [review]
Good job

**Review by allebd — COMMENTED** [review]
good one

**Review by allebd — APPROVED** [review]

**Review by KvNGCzA — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (Adekoreday) [reply .travis.yml]: treated accordingly

File: .travis.yml (hunk 3 above)
Comment (Adekoreday) [reply .travis.yml:8]: seen thanks

File: README.md (hunk 4 above)
Comment (Adekoreday) [reply README.md:7]: ok seen added

File: .travis.yml (hunk 6 above)
Comment (Adekoreday) [reply .travis.yml]: will rectify it
I don't think i can encrypt the token since I don't have access to the travis account
done

**Pull request merged**